core/common/src/ClipStyle.ts

```diff
@@ -97,6 +98,102 @@ export class CutStyle {
+  /** Color to apply to intersection of geometry and clip planes, default white */
+  color?: RgbColorProps;
+  /** Number of pixels to be considered intersecting the clip plane, default 1 */
+  width?: number;
+}
+
+/** As part of a [[ClipStyle]], describes how to colorize geometry intersecting the clip planes.
+ * @note Edges are highlighted only if [[ClipStyle.ClipIntersectionStyle]] is `true`.
+ * @public
+ * @extensions
+ */
+export class ClipIntersectionStyle {
+  /** Color to apply to intersection of geometry and clip planes, default white */
+  public readonly color: Readonly<RgbColor>;
+  /** Number of pixels to be considered intersecting the clip plane, default 1 */
+  public readonly width: number;
+
+  private constructor(color: RgbColor = RgbColor.fromColorDef(ColorDef.white), width: number = 1) {
+    this.color = color;
+    this.width = width;
+  }
+  /** Create a highlight  from its components. */
+  public static create(color?: RgbColor, width?: number): ClipIntersectionStyle {
+    if (!color && !width)
+      return this.defaults;
+
+    return new ClipIntersectionStyle(color, width);
+  }
+
+  public static readonly defaults = new ClipIntersectionStyle();
+
+  public static fromJSON(props?: ClipIntersectionStyleProps): ClipIntersectionStyle {
+    if (props === undefined) {
+      return ClipIntersectionStyle.defaults;
+    }
+
+    const color = props.color ? RgbColor.fromJSON(props.color) : RgbColor.fromColorDef(ColorDef.white);
+    const width = props.width ? props.width : 1;
+    return new ClipIntersectionStyle(color, width);
+  }
+
+  /** The JSON representation of this style. It is `undefined` if this style matches the defaults. */
+  public toJSON(): ClipIntersectionStyleProps | undefined {
+    const props: ClipIntersectionStyleProps = {};
+
+    if (this.matchesDefaults) {
+      return undefined;
+    }
+
+    if (this.color)
+      props.color = this.color.toJSON();
+
+    if (this.width)
+      props.width = this.width;
+
+    return props;
+  }
+
+  public get matchesDefaults(): boolean {
+    if (this === ClipIntersectionStyle.defaults)
+      return true;
+
+    return !this.color && !this.width;
+  }
+}
+
+/** Wire format describing a [[ClipStyle]].
+ * @see [[DisplayStyleSettingsProps.clipStyle]].
+ * @public
+ * @extensions
+ */
+export interface ClipStyleArgs {
+  /** If `true`, geometry will be produced at the clip planes in a 3d view.
+   * - Solids (closed volumes) will produce facets on the clip planes.
+   * - Other surfaces will produce line strings representing the edges of the surface at the clip planes.
+   * @note Cut geometry will only be produced for element geometry - not for, e.g., terrain or reality models.
+   */
+  produceCutGeometry: boolean;
+  /** If 'true', intersection of geometry and clip planes will be colorized */
+  colorizeIntersection: boolean;
+  /** Controls aspects of how the cut geometry is displayed, if [[produceCutGeometry]] is `true`. */
+  cutStyle: CutStyle;
```

Comment:
```suggestion
  produceCutGeometry?: boolean;
  /** If `true`, intersection of geometry and clip planes will be colorized */
  colorizeIntersection?: boolean;
  /** Controls aspects of how the cut geometry is displayed, if [[produceCutGeometry]] is `true`. */
  cutStyle?: CutStyle;
```

Callers should only have to supply arguments they care about. Use defaults for anything not supplied.